pkg/k8shandler/statefulset.go

```diff
@@ -134,77 +134,89 @@ func (node *statefulSetNode) waitForNodeLeaveCluster() (error, bool) {
 }
 
 func (node *statefulSetNode) setPartition(partitions int32) error {
+
+	desired := node.self.DeepCopy()
+
 	nretries := -1
 	retryErr := retry.RetryOnConflict(retry.DefaultRetry, func() error {
 		nretries++
-		if getErr := sdk.Get(&node.self); getErr != nil {
```

Comment:
i think we want to keep this as `&node.self` since we need to actually update the node here